### PR TITLE
WDR-40 inheritance

### PR DESCRIPTION
**Pull request by EarthmanT**

### CODE REVIEW

File: cloudify_starlingx/decorators.py [hunk 1]
```diff
@@ -41,7 +41,9 @@ def wrapper_inner(**kwargs):
             # node context
             ctx_node = resolve_ctx(ctx)
             client_config = ctx_node.node.properties.get('client_config')
-            cafile, cafilename = handle_cert_in_config(client_config)
+            cacert, cafile, cafilename = handle_cert_in_config(client_config)
```
Comment (AdarShaked) [inline cloudify_starlingx/decorators.py:44]: Are you using cafile and cafilename?

File: cloudify_starlingx/utils.py [hunk 2]
```diff
@@ -248,6 +258,26 @@ def update_deployment_site(deployment_id, site_name, rest_client):
...
+    :param deployment_id:
+    :param rest_client:
+    :return:
+    """
+    nodes = []
+    for node in rest_client.nodes.list(deployment_id=deployment_id,
+                                       _includes=['properties']):
+        if CONTROLLER_TYPE in node.type_hierarchy:
+            nodes.append(node)
+    if nodes:
```
Comment (AdarShaked) [inline cloudify_starlingx/utils.py:275]: if there should be only one controller maybe raise an error if the size of nodes >1

File: cloudify_starlingx/utils.py [hunk 3]
```diff
@@ -560,9 +590,10 @@ def handle_cert_in_config(client_config):
             cacert_as_file(cacert)
         client_config['os_cacert'] = cafilename
     else:
+        cacert = None
         cafile = None
         cafilename = None
-    return cafile, cafilename
+    return cacert, cafile, cafilename
```
Comment (AdarShaked) [inline cloudify_starlingx/utils.py:596]: I see you return None from this function on line 558 for example, and when you use it in decorator you expect 3 return values always. it's not a problem?

File: cloudify_starlingx/workflows/discover.py [hunk 4]
```diff
@@ -133,13 +136,22 @@ def generate_deployment_id(subclouds_name):
     controller_node_instance = get_controller_node_instance(
         node_instance_id, node_id, ctx=ctx)
 
+    parent_deployment_capabilities = get_parent_deployment_capabilities(
+        deployment=get_deployment(ctx.deployment.id))
+    auth_url = parent_deployment_capabilities['wrcp-ip']
```
Comment (AdarShaked) [inline cloudify_starlingx/workflows/discover.py:141]: those fields are always in the capabilities?

File: cloudify_starlingx/utils.py (hunk 3 above)
Comment (EarthmanT) [reply cloudify_starlingx/utils.py:596]: Nope

File: cloudify_starlingx/workflows/discover.py (hunk 4 above)
Comment (EarthmanT) [reply cloudify_starlingx/workflows/discover.py:141]: Yes

File: cloudify_starlingx/utils.py (hunk 2 above)
Comment (EarthmanT) [reply cloudify_starlingx/utils.py:275]: The blueprint is part of the package so they will always be using our blueprint.

File: cloudify_starlingx/decorators.py (hunk 1 above)
Comment (EarthmanT) [reply cloudify_starlingx/decorators.py:44]: yes

**Review by AdarShaked — APPROVED** [review]

**Pull request merged**